wordpress-mobile/WordPress-iOS
# Notification Service Extension incorporates FormattableContent.

Fixes #9936. A few changes were required to incorporate the existing `FormattableContent` component for use in the notification service extension. The distinct commit messages provide an idea of the changes, but in summary:

- A few attributes from `Notification` were promoted from `Notification` to `NotificationIdentifiable` (i.e., now `Notifiable`).
- Some tidying of `FormattableContent`, including correction of typos, removal of superfluous spaces, function visibility, etc.
- Modified Target Membership of some `FormattableContent` pieces. Additionally, a no-op (for now) `FormattableContentActionParser` was created to clarify the boundaries architecturally.
- Modified `Podfile` to be more clear about dependencies.
- The service extension now sets the title of the notification to the `alert` field of the `aps` node in the APNS payload. The body is set to the plain-text version of the notification block, as decoded via `FormattableContent`.

To test:

- The refactoring is best exercised by verifying that existing test pass.
- It is possible to test the push notification aspect manually; a sample test payload for use with _Pusher_ can be found [here](https://github.com/wordpress-mobile/WordPress-iOS/pull/9924). The one distinction is that the `note_id` field should map to a valid notification. Otherwise, the push will refrain from modifying the content.
